src/main/java/it/smartcommunitylab/aac/webauthn/service/WebAuthnUserAccountService.java

```diff
@@ -0,0 +1,167 @@
+        this.credentialRepository = credentialRepository;
+    }
+
+    @Transactional(readOnly = true)
+    public WebAuthnUserAccount getAccount(long id) throws NoSuchUserException {
+        WebAuthnUserAccount account = accountRepository.findOne(id);
+        if (account == null) {
+            throw new NoSuchUserException();
+        }
+        return accountRepository.detach(account);
+    }
+
+    @Transactional(readOnly = true)
+    public List<WebAuthnUserAccount> getAccounts(String userId) throws NoSuchUserException {
+        List<WebAuthnUserAccount> accounts = accountRepository.findBySubject(userId);
+        List<WebAuthnUserAccount> result = new LinkedList<>();
+        for (WebAuthnUserAccount account : accounts) {
+            result.add(accountRepository.detach(account));
+        }
+        if (result.isEmpty()) {
+            throw new NoSuchUserException();
+        }
+        return result;
+    }
+
+    @Transactional(readOnly = true)
+    public WebAuthnUserAccount find(long id) {
+        WebAuthnUserAccount account = accountRepository.findOne(id);
+        if (account == null) {
+            return null;
+        }
+        return accountRepository.detach(account);
+    }
+
+    @Transactional(readOnly = true)
+    public List<WebAuthnUserAccount> findBySubjectAndRealm(String subject, String realm) throws NoSuchUserException {
+        List<WebAuthnUserAccount> accounts = accountRepository.findBySubjectAndRealm(subject, realm);
+        List<WebAuthnUserAccount> result = new LinkedList<>();
+        for (WebAuthnUserAccount account : accounts) {
+            result.add(accountRepository.detach(account));
+        }
+        if (result.isEmpty()) {
+            throw new NoSuchUserException();
+        }
+        return result;
+    }
+
+    @Transactional(readOnly = true)
+    public WebAuthnUserAccount findByProviderAndUsername(String provider, String username) throws NoSuchUserException {
+        WebAuthnUserAccount account = accountRepository.findByProviderAndUsername(provider, username);
+        if (account == null) {
+            throw new NoSuchUserException();
+        }
+        return accountRepository.detach(account);
+    }
+
+    @Transactional(readOnly = true)
+    public WebAuthnUserAccount findByProviderAndSubject(String provider, String subject) throws NoSuchUserException {
+        WebAuthnUserAccount account = accountRepository.findByProviderAndSubject(provider, subject);
+        if (account == null) {
+            throw new NoSuchUserException();
+        }
+        return accountRepository.detach(account);
+    }
+
+    @Transactional(readOnly = true)
+    public WebAuthnUserAccount findByCredentialId(ByteArray credentialId) {
+        WebAuthnCredential cred = credentialRepository.findByCredentialId(credentialId.getBase64());
+        if (cred == null) {
+            return null;
+        }
+        return accountRepository.detach(cred.getParentAccount());
```

Comment:
you need to resolve the relation here by fetching the accountId from credentials and then looking into accountRepository